tests/element_test.py

```diff
@@ -108,7 +115,5 @@ def test_lens_propagate_fail():
         )
 
 def trigger_prop_fail(prop_func, pulse):
-    test_utils.trigger_exception_test(
-        prop_func,
-        [pulse, 'should fail']
-    )
+    with pkexcept(element.ElementException):
+        prop_func(pulse, 'should fail')
```

Comment:
"should fail" is not ideal. You might say "should raise". You can also say something like this:

with pkexcept(element.ElementException, 'Invalid element="should raise" should have raised')

This will print an explanatory error message which is also easy to find in the test. Reading "prop_func(pulse, 'should fail')" isn't intuitively obvious why it should raise.